### PR TITLE
Library widget revamp

### PR DESCRIPTION
**Pull request by partouf**
Basic working version for testing.

Todo:
 * [x] Layout not stable when selecting, searching and favoriting
 * [x] Needs a fixed height (but still scroll)
 * [x] Needs dark-mode attention
 * [x] Focus search box when dialog opens
 * [x] Search when typing
 * [x] Hint in the search box (search... / filter...)
 * [x] Needs some tweaks when it comes to hidden and aliased libraries
 * [x] Try to tweak mobile a bit so its more usable (also in portrait mode)
 * [x] Highlight Libraries button on compiler pane
 * [x] Make work for Execution and Conformance

Todo properties file
 * [x] Fill in more descriptions
 * [x] Make sure every lib has a website url

### CODE REVIEW

**Comment by partouf** [comment]
tried to set modal height to a certain minimum, but caused various issues, leaving height as is (so dynamic depending on how much there is to show)

File: static/libs-widget-ext.js [hunk 1]
```diff
@@ -0,0 +1,481 @@
...
+LibsWidgetExt.prototype.conjureUpExamples = function (result, lib) {
+    var examples = result.find('.lib-examples');
+    if (lib.examples.length > 0) {
+        var examplesHeader = $('<b>Examples</b>');
+        var examplesList = $('<ul />');
+        _.each(lib.examples, function (exampleId) {
+            var li = $('<li />');
+            examplesList.append(li);
+            var exampleLink = $('<a>Example</a>');
+            exampleLink.attr('href', '/z/' + exampleId);
```
Comment (mattgodbolt) [inline static/libs-widget-ext.js:209]: can we add a target="_blank" here please?

File: static/libs-widget-ext.js [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+        this.markLibrary(libId, versionId, false);
+        libDiv.remove();
+        this.showSelectedLibs();
+        this.onChangeCallback();
+    }, this));
+
+    return libDiv;
+};
+
+LibsWidgetExt.prototype.conjureUpExamples = function (result, lib) {
```
Comment (mattgodbolt) [inline static/libs-widget-ext.js:200]: :heart: the name :)

File: static/libs-widget-ext.js [hunk 3]
```diff
@@ -0,0 +1,481 @@
...
+    }
+};
+
+LibsWidgetExt.prototype.newSearchResult = function (libId, lib) {
+    var template = $('#lib-search-result-tpl');
+
+    var result = $(template.children()[0].cloneNode(true));
+    result.find('.lib-name').html(lib.name);
+    result.find('.lib-description').html(lib.description ? lib.description : '&nbsp;');
+    result.find('.lib-website-link').attr('href', lib.url ? lib.url : '#');
```
Comment (mattgodbolt) [inline static/libs-widget-ext.js:224]: ditto a target="_blank"?

Or maybe, this one should be _blank, but the "examples" should replace the current view? I don't want folks to lose their in-progress code, is all...probably both should make a new tab.

**Review by mattgodbolt — APPROVED** [review]
Still looks ace! As discussed maybe a tiny bit of text to explain users need to pick a version

File: static/libs-widget-ext.js (hunk 3 above)
Comment (apmorton) [reply static/libs-widget-ext.js:224]: these external links also need `rel="noopener"`

**Pull request merged**